static/library/slider_component.ts

```diff
@@ -249,47 +319,98 @@ export class DatacommonsSliderComponent extends LitElement {
       `;
     }
 
+    const startDate = this._dates[0];
+    const endDate = this._dates[this._dates.length - 1];
+    // Normalized slider value as a percent between 0 and 100
+    const normalizedSliderValue =
+      this._showLatestAvailableData || this._dates.length <= 1
+        ? 100
+        : (this._value / (this._dates.length - 1)) * 100;
+
+    const dateText = this.getDateText();
+    const endDateText = this.getEndDateText();
+    const lastDateIndex = this._dates.length - 1;
+    const isHighestCoverageDate = dateText === this._highestCoverageDate;
+
+    // Text to show under range slider button.
+    // Shows asterisk if showing latest available data or date of highest
+    // coverage
+    const sliderLabelText = `${
+      this._showLatestAvailableData ? endDateText : dateText
+    }${isHighestCoverageDate || this._showLatestAvailableData ? "*" : ""}`;
     return html`
       <div class="container" part="container">
         ${this.header
           ? html`<h4 part="header">${this.header}</h4>`
           : this.defaultHeader()}
-        <div class="row">
-          <div class="label">${this._dates[0]}</div>
-          <input
-            class="slider"
-            max="${this._dates.length}"
-            min="0"
-            title="${this.getValueText()}"
-            type="range"
-            value="${this._value}"
-            @change=${this.onSliderChange}
-            @input=${this.onSliderInput}
-          />
-          <div class="label">${MOST_RECENT_TEXT}</div>
+        <div
+          class="row slider ${this._showLatestAvailableData ? "disabled" : ""}"
+        >
+          <div class="label">${startDate}</div>
+          <div class="slider-control">
+            <input
+              class="slider"
+              max="${lastDateIndex}"
+              min="0"
+              title="${dateText}"
+              type="range"
+              .value="${this._showLatestAvailableData
```

Comment:
should this field be `value`?

Reply:
This is a nuance of the HTML input type="range" component.. apparently you have to set the value in javascript in order to see the change, just setting the attribute on the dom isn't enough: https://stackoverflow.com/questions/42460990/html-range-input-doesnt-update-its-value. `.value` is a shortcut for doing this in lit